File: src/main/java/com/ject/studytrip/studylog/application/dto/StudyLogDailyMissionInfo.java
package com.ject.studytrip.studylog.application.dto;

import com.ject.studytrip.mission.application.dto.DailyMissionInfo;
import com.ject.studytrip.studylog.domain.model.StudyLogDailyMission;

public record StudyLogDailyMissionInfo(
        Long studyLogDailyMissionId, DailyMissionInfo dailyMissionInfo) {
    public static StudyLogDailyMissionInfo from(StudyLogDailyMission studyLogDailyMission) {
        return new StudyLogDailyMissionInfo(
                studyLogDailyMission.getId(),
                DailyMissionInfo.from(studyLogDailyMission.getDailyMission()));
    }
}

File: src/main/kotlin/com/ject/studytrip/studylog/application/dto/StudyLogDailyMissionInfo.kt
package com.ject.studytrip.studylog.application.dto

import com.ject.studytrip.global.util.EntityExtensions.requireId
import com.ject.studytrip.mission.application.dto.DailyMissionInfo
import com.ject.studytrip.studylog.domain.model.StudyLogDailyMission

data class StudyLogDailyMissionInfo(
    val studyLogDailyMissionId: Long,
    val dailyMissionInfo: DailyMissionInfo,
) {
    companion object {
        fun from(studyLogDailyMission: StudyLogDailyMission): StudyLogDailyMissionInfo =
            StudyLogDailyMissionInfo(studyLogDailyMission.id.requireId(), DailyMissionInfo.from(studyLogDailyMission.dailyMission))
    }
}
